test/ring/middleware/oauth2_test.clj

```diff
@@ -231,6 +245,29 @@
         (is (approx-eq (-> 3600 time/seconds time/from-now)
                        (-> response :session ::oauth2/access-tokens :test :expires)))))))
 
+(defn openid-response-with-code-verifier [req]
+  {:status 200
+   :headers {"Content-Type" "application/json"}
+   :body (cheshire/generate-string
+          {:access_token "defdef"
+           :expires_in 3600
+           :refresh_token "ghighi"
+           :id_token "abc.def.ghi"
+           :code_verifier (-> req :body slurp codec/form-decode (get "code_verifier"))})})
+
+(deftest test-openid-response-with-code-verifier
+  (fake/with-fake-routes
+   {"https://example.com/oauth2/access-token" openid-response-with-code-verifier}
+
+   (testing "valid state"
+            (let [verifier (oauth2/code-verifier)
```

Comment:
`testing` indentation not correct, should be indented two spaces. Also take care to ensure lines are within 80 characters in this test.